javascript/javascript3/week1/week1weatherapp.js

```diff
@@ -0,0 +1,50 @@
+const userCity = document.getElementById("userCity");
+const userButton = document.getElementById("userButton");
+const weatherIcon = document.getElementById("weather_icon");
+let weatherUrl;
+
+//Check for input city
+userButton.addEventListener("click", getCurrentWeatherByCityName);
+  function getCurrentWeatherByCityName() 
+  {
+    const cityName = userCity.value;
+    const appkey = "3cde58a1dd84fa31e343be2e7ebcfb24";
+       if (!cityName) 
+        {
+          alert("Please enter city");
+        }
+        else
+        {
+            weatherUrl = `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&appid=${appkey}`;
+            getWeather();
+        }
+  }
+  //getting data and display in browser
+  function getWeather()
+  {
+    fetch(weatherUrl)
```

Comment:
In general, it is a good idea to add code for error handling. For example in cases when the requests fails. fetch(url) .then(response) => {} .catch((error) => { console.error('Error:', error); });